### DISCUSSION TITLE
How to run android example from emulator?

### DISCUSSION THREAD

**Discussion by wusyong** [BrainiumLLC/cargo-mobile#61] [Q&A] [answered]
Sorry if this feels like a noob question. But I would like to know if it's possible to run a demo through android emulator.
`cargo android run` seems to need a actual device and I'm have none at the moment.

**Comment by zeerooth** [comment, marked as answer]
Hello!
To run your app through an emulator you need to open the project in Android Studio with `cargo android open` and then follow these instructions:
https://developer.android.com/studio/run/emulator
This should hopefully work for you

**Reply by wusyong** [reply]
Thanks! I found it's I didn't set the gradle version. But I did follow the link you provide and run the app in emulator successfully!
